[ENH] Added read/write method for network object (#651)
jonescompneurolab/hnn-core: doc/network_file_desc.rst

:orphan:

.. _network_file_content:

====================
Network File Content

> ====================

Reviewer: can you add some more information what the file structure is like? what is the file format etc? we need to provide some more information for users if they open the file using other tools and also for documentation

File Content Description
------------------------

>     object_type : str

Reviewer: you have an extra tab level which is making the rendering appear as if it's quoted:

https://output.circle-artifacts.com/output/job/2e226fdd-4b37-4b45-84b2-23faa4488796/artifacts/0/html/network_file_desc.html

:orphan:

.. _network_file_content:

====================
Network File Content
====================

File Content Description
------------------------

> hdf5 is the file format used for storing the Network object. The network is stored in a layered format. The first layer consists of the network attributes.

Reviewer 2: ```suggestion
hdf5 is the file format used for storing the Network object. The network is stored in a multi-level format. The first level consists of the network attributes.
```

Using "layer" as the terminology might get confusing, as there is the biologically "layers" of the model

The attributes of the network are then broken down until the network can be representated as a collection of key value pairs. For example - cell_types is a network

> attribute therefore in the first layer. The description of each cell type is in layer 2. Each cell has various sections. The description of a section is in layer 3.

Reviewer 2: ```suggestion
attribute therefore in the first level. The description of each cell type is in level 2. Each cell has various sections. The description of a section is in level 3.
```

Below are the attributes of the network.

object_type : str
    Type of object (Network) saved.

> N_pyr_x : int
>     Nr of cells (x).
> N_pyr_y : int
>     Nr of cells (y).

Reviewer 2: ```suggestion
N_pyr_x : int
    Number of cells (x).
N_pyr_y : int
    Number of cells (y).
```
